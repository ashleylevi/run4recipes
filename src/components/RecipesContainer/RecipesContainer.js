import React, { Component } from 'react';
import { fetchPastaRecipes, fetchPotatoRecipes, fetchBreadRecipes } from '../../utils/apiCalls';
import { loadAllRecipes, loadPastaRecipes, loadPotatoRecipes, loadBreadRecipes } from '../../actions/index';
import { connect } from 'react-redux';
import './RecipesContainer.css'

class RecipesContainer extends Component {
  constructor() {
    super()
  }

  async componentDidMount() {
      const pastaRecipes = await fetchPastaRecipes()
      const potatoRecipes = await fetchPotatoRecipes()
      const breadRecipes = await fetchBreadRecipes()
      this.props.loadAllRecipes(pastaRecipes, potatoRecipes, breadRecipes)    
  }

  render() {
    return (
      <div className="recipes-container">

      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  allRecipes: state.allRecipes
})

export const mapDispatchToProps = (dispatch) => ({
  loadAllRecipes: (recipes1, recipes2, recipes3) => dispatch(loadAllRecipes(recipes1, recipes2, recipes3))
})

export default connect(mapStateToProps, mapDispatchToProps)(RecipesContainer)