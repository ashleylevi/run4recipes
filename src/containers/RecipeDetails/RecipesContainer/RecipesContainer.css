.recipes-container {
  display: grid;
    grid-template-columns: repeat(auto-fit, 254px);
    grid-gap: 20px;
    justify-content: center;
    margin: 30px auto;
}

.no-favorites {
  color: red;
  font-size: 50px;
}