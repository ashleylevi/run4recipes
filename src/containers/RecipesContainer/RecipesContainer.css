.recipes-container {
  display: grid;
    grid-template-columns: repeat(auto-fit, 254px);
    grid-gap: 20px;
    justify-content: center;
    margin: 30px auto;
}

.no-favorites {
  font-size: 25px;
  text-align: center; 
  color: #32636E;
  font-weight: 700;
  margin-top: 50px;
}