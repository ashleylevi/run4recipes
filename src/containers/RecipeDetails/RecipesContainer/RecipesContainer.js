import React, { Component } from 'react';
import { connect } from 'react-redux';
import './RecipesContainer.css';
import { uid } from 'react-uid';
import { RecipeCard } from '../RecipeCard/RecipeCard';
import { fetchPastaThunk } from '../../thunks/loadPastaRecipes';
import { fetchPotatoThunk } from '../../thunks/loadPotatoRecipes';
import { fetchBreadThunk } from '../../thunks/loadBreadRecipes';
import { Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

class RecipesContainer extends Component {
  constructor() {
    super()
    this.state = {
      updateFavorites: false
    }
  }

  async componentDidMount() {
    this.props.fetchPastaThunk()
    this.props.fetchPotatoThunk()
    this.props.fetchBreadThunk()
  }

  updateFavorites = () => {
    this.setState({
      updateFavorites: true
    })
  }

  render() {
    const { pastaRecipes, potatoRecipes, breadRecipes, match } = this.props;
    let recipes;
    let recipesToDisplay;
    let str = match.path.substring(1);
    if (match.path !== '/' && match.path !== '/favorites') {
      recipes = this.props[`${str}Recipes`];
      recipesToDisplay = recipes.map((recipe) => {
        return (<Link to={`/${str}/${recipe.name}`}><RecipeCard recipe={recipe} key={uid(recipe)} updateFavorites={this.updateFavorites} /></Link>)
      })
    } else if (match.path === '/favorites') {
      const favorites = JSON.parse(localStorage.getItem('faves'))
        if (!favorites.length) {
          recipesToDisplay = <p className="no-favorites">You have no favorites</p>
        } else {
          recipesToDisplay = favorites.map((recipe) => {
            return <RecipeCard recipe={recipe} key={uid(recipe)} updateFavorites={this.updateFavorites}/>
          }) 
        }
    } else {
      recipes = [...pastaRecipes, ...breadRecipes, ...potatoRecipes];
      recipesToDisplay = recipes.map((recipe) => {
        return (<Link to={`/${recipe.category}/${recipe.name}`}><RecipeCard recipe={recipe} key={uid(recipe)} updateFavorites={this.updateFavorites} /></Link>)
      })
    }

    return (
      <div className="recipes-container">
        { recipesToDisplay }
        {
          favorites.length === 0 && <h2 className="no-favorites">You Have No Favorites</h2>
        }
      </div>
    )
  }
}


export const mapStateToProps = (state) => ({
  pastaRecipes: state.pastaRecipes,
  potatoRecipes: state.potatoRecipes,
  breadRecipes: state.breadRecipes
})

export const mapDispatchToProps = (dispatch) => ({
  fetchPastaThunk: () => dispatch(fetchPastaThunk()),
  fetchPotatoThunk: () => dispatch(fetchPotatoThunk()),
  fetchBreadThunk: () => dispatch(fetchBreadThunk())
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(RecipesContainer))

RecipesContainer.propTypes = {
  pastaRecipes: PropTypes.array,
  potatoRecipes: PropTypes.array,
  breadRecipes: PropTypes.array,
  fetchPastaThunk: PropTypes.func,
  fetchPotatoThunk: PropTypes.func,
  fetchBreadThunk: PropTypes.func
};