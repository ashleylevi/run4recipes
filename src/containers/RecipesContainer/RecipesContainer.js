import React, { Component } from 'react';
import { connect } from 'react-redux';
import './RecipesContainer.css';
import { uid } from 'react-uid';
import { RecipeCard } from '../../components/RecipeCard/RecipeCard';
import { fetchPastaThunk } from '../../thunks/loadPastaRecipes';
import { fetchPotatoThunk } from '../../thunks/loadPotatoRecipes';
import { fetchBreadThunk } from '../../thunks/loadBreadRecipes';
import { Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

class RecipesContainer extends Component {
  constructor() {
    super()
    this.state = {
      updateFavorites: false
    }
  }

  async componentDidMount() {
    this.props.fetchPastaThunk()
    this.props.fetchPotatoThunk()
    this.props.fetchBreadThunk()
  }

  updateFavorites = () => {
    this.setState({
      updateFavorites: true
    })
  }

  render() {
    const { pastaRecipes, potatoRecipes, breadRecipes, match } = this.props;
    let recipes;
    let recipesToDisplay;
    let str = match.path.substring(1);
    if (match.path !== '/' && match.path !== '/favorites') {
      console.log('hi')
      recipes = this.props[`${str}Recipes`];
      recipesToDisplay = recipes.map((recipe) => {
        return (<Link to={`/${str}/${recipe.name}`}><RecipeCard recipe={recipe} key={uid(recipe)} updateFavorites={this.updateFavorites} /></Link>)
      })
    } else if (match.path === '/favorites') {
      const favorites = JSON.parse(localStorage.getItem('faves'))
        if (!favorites.length) {
          recipesToDisplay = "You have no favorites!"
        } else {
          recipesToDisplay = favorites.map((recipe) => {
            return <RecipeCard recipe={recipe} key={uid(recipe)} updateFavorites={this.updateFavorites}/>
          }) 
        }
    } else {
      recipes = [...pastaRecipes, ...breadRecipes, ...potatoRecipes];
      recipesToDisplay = recipes.map((recipe) => {
        return (<Link to={`/${recipe.category}/${recipe.name}`}><RecipeCard recipe={recipe} key={uid(recipe)} updateFavorites={this.updateFavorites} /></Link>)
      })
    }

    return (
      <div className="recipes-container">
        { recipesToDisplay }
      </div>
    )
  }
}


export const mapStateToProps = (state) => ({
  pastaRecipes: state.pastaRecipes,
  potatoRecipes: state.potatoRecipes,
  breadRecipes: state.breadRecipes
})

export const mapDispatchToProps = (dispatch) => ({
  fetchPastaThunk: () => dispatch(fetchPastaThunk()),
  fetchPotatoThunk: () => dispatch(fetchPotatoThunk()),
  fetchBreadThunk: () => dispatch(fetchBreadThunk())
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(RecipesContainer))

RecipesContainer.propTypes = {
  pastaRecipes: PropTypes.array,
  potatoRecipes: PropTypes.array,
  breadRecipes: PropTypes.array,
  fetchPastaThunk: PropTypes.func,
  fetchPotatoThunk: PropTypes.func,
  fetchBreadThunk: PropTypes.func
};