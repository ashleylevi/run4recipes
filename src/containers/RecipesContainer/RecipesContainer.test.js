import { mapStateToProps,
   mapDispatchToProps } from './RecipesContainer';
import RecipesContainer from './RecipesContainer';
import React from 'react';
import { shallow } from 'enzyme';

jest.mock('../../thunks/loadPastaRecipes')
jest.mock('../../thunks/loadPotatoRecipes')
jest.mock('../../thunks/loadBreadRecipes')
jest.mock('react-uid', () => ({
  uid: jest.fn(() => 1)
}))

describe('RecipesContainer Component', () => {
  let wrapper;
  let mockRecipe;
 
  
  beforeEach(() => {
    wrapper = shallow(<RecipesContainer />)
  })
    mockRecipe = {
      name: "Spaghetti and meatballs",
      image: "https://image.png",
      ingredients: ['spaghetti', 'meatballs'],
      category: 'pasta'
    }
  
  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should match the snapshot if favorites ', () => {
    const mockFavoritesPath = {
      path: 'favorites',
      params: {
        name: 'baked potato'
      }
    }
    localStorage.setItem('faves', JSON.stringify([mockRecipe]))
    const wrapper = shallow(<RecipesContainer recipe={mockRecipe} updateFavorites={jest.fn()} match={mockFavoritesPath}/>)
    expect(wrapper).toMatchSnapshot()
  })


  describe('mapStateToProps', () => {
    it('should create an object with everything in state', () => {
      const mockState = {
        pastaRecipes: [{}],
        potatoRecipes: [{}],
        breadRecipes: [{}],
        isLoading: false,
        searchValue: 'kimchi',
        isError: 'false'
      }

      const expected = {
        pastaRecipes: [{}],
        potatoRecipes: [{}],
        breadRecipes: [{}],
        isLoading: false,
        searchValue: 'kimchi',
        isError: 'false'
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a fetchPasta action', () => {
      const mockDispatch = jest.fn()
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchPastaThunk()
      expect(mockDispatch).toHaveBeenCalledWith('pasta thunk')
    })
    it('calls dispatch with a fetchPotato action', () => {
      const mockDispatch = jest.fn()
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchPotatoThunk()
      expect(mockDispatch).toHaveBeenCalledWith('potato thunk')
    })
    it('calls dispatch with a fetchBread action', () => {
      const mockDispatch = jest.fn()
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchBreadThunk()
      expect(mockDispatch).toHaveBeenCalledWith('bread thunk')
    })


   
    })
  })


  