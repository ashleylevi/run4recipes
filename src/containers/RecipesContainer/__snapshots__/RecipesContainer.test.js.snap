// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipesContainer Component Should match the snapshot if favorites  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Connect(RecipesContainer))
    match={
      Object {
        "params": Object {
          "name": "baked potato",
        },
        "path": "favorites",
      }
    }
    recipe={
      Object {
        "category": "pasta",
        "image": "https://image.png",
        "ingredients": Array [
          "spaghetti",
          "meatballs",
        ],
        "name": "Spaghetti and meatballs",
      }
    }
    updateFavorites={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`RecipesContainer Component matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Connect(RecipesContainer)) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
